### PR TITLE
fix parser for new integers

### PR DESCRIPTION
**Pull request by oxade**
## Motivation

Fixes [type tag parser missing new integers](https://github.com/move-language/move/issues/650#event-7748015930).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes.
## Test Plan

Added parser tests.

### CODE REVIEW

**Comment by oxade** [comment]
Failing test seems unrelated.
If anyone has the ability, please restart it.

File: language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt [hunk 1]
```diff
@@ -4,6 +4,11 @@ sandbox publish
 sandbox generate struct-layouts --module storage/0x00000000000000000000000000000001/modules/M2.mv --struct C --type-args u64
 sandbox generate struct-layouts --module storage/0x00000000000000000000000000000001/modules/M2.mv --struct C --type-args address
 sandbox generate struct-layouts --module storage/0x00000000000000000000000000000001/modules/M2.mv --struct C --type-args vector<u8>
+sandbox generate struct-layouts --module storage/0x00000000000000000000000000000001/modules/M2.mv --struct C --type-args u8
```
Comment (sblackshear) [inline language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt:7]: This is a somewhat indirect way to test the type tag parser... I would suggest adding some new unit tests to the bottom of `parser.rs` instead

**Review by sblackshear — CHANGES_REQUESTED** [review]

**Comment by sblackshear** [comment]
Thanks for fixing this!

File: language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt (hunk 1 above)
Comment (oxade) [reply language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt:7]: Very true, thank you. Fixed now.
But something weird/unrelated is causing CI failures at build setup stage.

**Review by sblackshear — APPROVED** [review]

File: language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt (hunk 1 above)
Comment (sblackshear) [reply language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt:7]: Thanks for adding the unit tests! Would you mind deleting these other tests?

Comment (oxade) [reply language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt]: Done.

Comment (oxade) [reply language/tools/move-cli/tests/sandbox_tests/generate_struct_layout/args.txt]: Done. Thanks for feedback. Still learning my way around the repo.

**Comment by oxade** [comment]
@sblackshear CI seems good now. Can you please merge?

**Review by tnowacki — APPROVED** [review]

**Pull request merged**